warden/x/warden/module/module.go

```diff
@@ -180,16 +185,53 @@ type ModuleInputs struct {
 	Config       *modulev1.Module
 	Logger       log.Logger
 
-	AccountKeeper types.AccountKeeper
-	BankKeeper    types.BankKeeper
-	IntentKeeper  types.IntentKeeper
+	AccountKeeper  types.AccountKeeper
+	BankKeeper     types.BankKeeper
+	FeeGrantKeeper types.FeegrantKeeper
+	IntentKeeper   types.IntentKeeper
+	TxConfig       client.TxConfig
 }
 
 type ModuleOutputs struct {
 	depinject.Out
 
-	WardenKeeper keeper.Keeper
-	Module       appmodule.AppModule
+	WardenKeeper  keeper.Keeper
+	Module        appmodule.AppModule
+	BaseAppOption runtime.BaseAppOption
+}
+
+func newAnteHandler(in ModuleInputs) (sdk.AnteHandler, error) {
+	if in.AccountKeeper == nil {
+		return nil, fmt.Errorf("account keeper is required for ante builder")
+	}
+
+	if in.BankKeeper == nil {
+		return nil, fmt.Errorf("bank keeper is required for ante builder")
+	}
+
+	if in.TxConfig == nil {
+		return nil, fmt.Errorf("tx config is required for ante builder")
+	}
+
+	if in.TxConfig.SignModeHandler() == nil {
+		return nil, fmt.Errorf("sign mode handler is required for ante builder")
+	}
+
+	anteHandler, err := ante.NewAnteHandler(
+		ante.HandlerOptions{
+			AccountKeeper:   in.AccountKeeper,
+			BankKeeper:      in.BankKeeper,
+			SignModeHandler: in.TxConfig.SignModeHandler(),
+			FeegrantKeeper:  in.FeeGrantKeeper,
+			SigGasConsumer:  authante.DefaultSigVerificationGasConsumer,
+			FeeBurnRatio:    math.LegacyNewDecWithPrec(3, 1), // 30%
```

Comment:
Do you know if we can get this number from the modules Params structure? This way it can get updated via governance proposals

Reply:
Yes, we definitely can. Good point, I'll implement it.